File: navigation/src/main/java/com/example/navigation/stepsEngine/field/CardTagField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.enums.Tag;
import com.example.navigation.stepsEngine.payment.FlowState;

public class CardTagField extends Field {

    @Override
    public String get(FlowState flowState) {
        final Tag tag = flowState.getCard().getTag();
        if (tag == null) {
            return null;
        } else {
            return tag.name();
        }
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/CardTagField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState

class CardTagField : Field() {
    override operator fun get(flowState: FlowState): String? {
        val tag = flowState.card.tag
        return tag?.name
    }

    override fun getId(): String {
        return FieldName.CARD_TAG
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}